spec/apis/v1/submission_comments_api_spec.rb

```diff
@@ -65,4 +82,102 @@ def preflight(preflight_params)
+          {
+            :controller => 'submission_comments_api',
+            :action => 'update',
+            :format => 'json',
+            :course_id => @course.id.to_s,
+            :assignment_id => @assignment.id.to_s,
+            :user_id => @student.id.to_s,
+            :id => @comment.to_param
+          },
+          {
+            :comment => "Goodbye world!"
+          },
+          {},
+          {:expected_status => 401})
+          expect(@comment.reload.comment).to eq("Hello world!")
+      end
+    end
+
+    context "user has permission to edit the comment" do
+      it "edits the submission comment" do
+        api_call(:put,
+          "/api/v1/courses/#{@course.id}/assignments/#{@assignment.id}/submissions/#{@student.id}/comments/#{@comment.id}",
+          {
+            :controller => 'submission_comments_api',
+            :action => 'update',
+            :format => 'json',
+            :course_id => @course.id.to_s,
+            :assignment_id => @assignment.id.to_s,
+            :user_id => @student.id.to_s,
+            :id => @comment.to_param
+          },
+          {
+            :comment => "Goodbye world!"
+          },
+          {},
+          {:expected_status => 200})
+          expect(@comment.reload.comment).to eq("Goodbye world!")
+      end
+    end
+  end
+
+  describe '#destroy' do
+    before :once do
+      submission_with_comment
+    end
+
+    context "user does not have the permission to delete the comment" do
+      it "does not delete the submission comment" do
+        @user = @student
+        api_call(:delete,
+          "/api/v1/courses/#{@course.id}/assignments/#{@assignment.id}/submissions/#{@student.id}/comments/#{@comment.id}",
+          {
+            :controller => 'submission_comments_api',
+            :action => 'destroy',
+            :format => 'json',
+            :course_id => @course.id.to_s,
+            :assignment_id => @assignment.id.to_s,
+            :user_id => @student.id.to_s,
+            :id => @comment.to_param
+          },
+          {},
+          {},
+          {:expected_status => 401})
+        expect(@submission.submission_comments.size()).to eq(1)
+      end
+    end
+
+    context "user has permission to delete the comment" do
+      it "deletes the submission comment" do
+        api_call(:delete,
+          "/api/v1/courses/#{@course.id}/assignments/#{@assignment.id}/submissions/#{@student.id}/comments/#{@comment.id}",
+          {
+            :controller => 'submission_comments_api',
+            :action => 'destroy',
+            :format => 'json',
+            :course_id => @course.id.to_s,
+            :assignment_id => @assignment.id.to_s,
+            :user_id => @student.id.to_s,
+            :id => @comment.to_param
+          },
+          {},
+          {},
+          {:expected_status => 200})
+        expect(@submission.submission_comments.size()).to eq(0)
```

Comment:
Nitpick comment (feel free to ignore this, or not make the change if you disagree):

```
expect(@submission.submission_comments).to be_empty
```

would be a bit more idiomatic